src/ui.rs

```diff
@@ -1,37 +1,41 @@
 use dioxus::prelude::*;
 
+#[derive(PartialEq, Clone, Copy)]
 pub enum Variant {
     Primary,
     Secondary,
     Neutral,
 }
 
-pub fn text_button(
-    text: &str,
-    onclick: impl FnMut(Event<MouseData>) + 'static,
-    class: &str,
-    variant: Variant,
-) -> Element {
-    button(
-        rsx!({ text }),
-        onclick,
-        &["py-3 px-6", class].join(" "),
-        variant,
-    )
+#[derive(PartialEq, Props, Clone)]
+pub struct ButtonProps {
+    pub text: String,
+    pub onclick: EventHandler<MouseEvent>,
+    pub class: String,
+    pub variant: Variant,
 }
 
-pub fn button(
-    text: Element,
-    onclick: impl FnMut(Event<MouseData>) + 'static,
-    class: &str,
-    variant: Variant,
-) -> Element {
+#[component]
+pub fn TextButton(props: ButtonProps) -> Element {
```

Comment:
When using `#[component]` you can inline the props as if they were normal parameters without the need of a separate struct for the props:

```rs
#[component]
pub fn TextButton(text: String, onclick: EventHandler<MouseEvent>, class: String, variant: Variant) -> Element {
```

Or you can keep the `ButtonProps` struct and remove `#[component]`, both work the same way :)